jrcc-access-spring-boot-autoconfigure/src/main/java/ca/bc/gov/open/jrccaccess/autoconfigure/plugins/sftp/SftpDocumentInput.java

```diff
@@ -67,4 +69,14 @@ private String getContent(Message<InputStream> message) throws IOException {
         return stringBuilder.toString();
     }
 
+    private String getFilename(Message<InputStream> message) throws IOException{
+        MessageHeaders messageHeaders = message.getHeaders();
+        Object filenameObj = messageHeaders.get("file_remoteFile");
+        if(filenameObj == null ){
+            IOException exception = new IOException("corrupted SFTP header. Filename is required.");
```

Comment:
I think it's right to throw an exception here, but no IOexception, IOexception are when you interact with a file, here there is a missing keyvalue in the message header, I would create a custom exception FileNameMissingException that extends DocumentException ... what do you think

Reply:
ok. I will do that.